File: leetcode/medium/48. Rotate Image/solution.js
/**
 * @param {number[][]} matrix
 * @return {void} Do not return anything, modify matrix in-place instead.
 */
function rotate(matrix) {   
    const size = matrix.length;
    for (let i = 0; i < size; i += 1) {
        for (let j = i + 1; j < size; j += 1) {
            [matrix[i][j], matrix[j][i]] = [matrix[j][i], matrix[i][j]];
        }
    }
    for (const m of matrix) m.reverse();
};
rotate([[5,1,9,11],[2,4,8,10],[13,3,6,7],[15,14,12,16]])

File: leetcode/medium/48. Rotate Image/solution.cc
#include <vector>

using namespace std;

class Solution {
public:
    void rotate(vector<vector<int>>& m) {
        int size = m.size(), temp;
        for (int i = 0; i < size / 2; i++) {
            for (int j = i; j < size - 1 - i; j++) {
                temp = m[i][j];
                m[i][j] = m[size - 1 - j][i];
                m[size - 1 - j][i] = m[size - 1 - i][size - 1 - j];
                m[size - 1 - i][size - 1 - j] = m[j][size - 1 - i];
                m[j][size - 1 - i] = temp;
            }
        }
    }
};